FirebaseAppDistribution/Tests/Unit/FIRFADApiServiceTests.m

```diff
@@ -0,0 +1,411 @@
+
+  [FIRFADApiService
+      generateAuthTokenWithCompletion:^(NSString *_Nullable identifier,
+                                        FIRInstallationsAuthTokenResult *_Nullable authTokenResult,
+                                        NSError *_Nullable error) {
+        XCTAssertNotNil(authTokenResult);
+        XCTAssertNotNil(identifier);
+        XCTAssertNil(error);
+        XCTAssertEqual(identifier, self->_mockInstallationId);
+        XCTAssertEqual([authTokenResult authToken], self -> _mockAuthToken);
+        [expectation fulfill];
+      }];
+
+  [self waitForExpectations:@[ expectation ] timeout:5.0];
+}
+
+- (void)testGenerateAuthTokenWithCompletionAuthTokenFailure {
+  [self mockInstallationAuthCompletion:nil
+                                 error:[NSError errorWithDomain:kFakeErrorDomain
+                                                           code:1
+                                                       userInfo:@{}]];
+  [self mockInstallationIdCompletion:_mockInstallationId error:nil];
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Generate auth token fails to generate auth token."];
+
+  [FIRFADApiService
+      generateAuthTokenWithCompletion:^(NSString *_Nullable identifier,
+                                        FIRInstallationsAuthTokenResult *_Nullable authTokenResult,
+                                        NSError *_Nullable error) {
+        XCTAssertNil(identifier);
+        XCTAssertNil(authTokenResult);
+        XCTAssertNotNil(error);
+        XCTAssertEqual(error.code, FIRFADApiTokenGenerationFailure);
+        [expectation fulfill];
+      }];
+
+  [self waitForExpectations:@[ expectation ] timeout:5.0];
+}
+
+- (void)testGenerateAuthTokenWithCompletionIDFailure {
+  [self mockInstallationAuthCompletion:_mockInstallationToken error:nil];
+  [self mockInstallationIdCompletion:nil
+                               error:[NSError errorWithDomain:kFakeErrorDomain
+                                                         code:1
+                                                     userInfo:@{}]];
+
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Generate auth token fails to find ID."];
+
+  [FIRFADApiService
+      generateAuthTokenWithCompletion:^(NSString *_Nullable identifier,
+                                        FIRInstallationsAuthTokenResult *_Nullable authTokenResult,
+                                        NSError *_Nullable error) {
+        XCTAssertNil(identifier);
+        XCTAssertNil(authTokenResult);
+        XCTAssertNotNil(error);
+        XCTAssertEqual([error code], FIRFADApiInstallationIdentifierError);
+        [expectation fulfill];
+      }];
+
+  [self waitForExpectations:@[ expectation ] timeout:5.0];
+}
+
+- (void)testFetchReleasesWithCompletionSuccess {
+  NSHTTPURLResponse *fakeResponse = OCMClassMock([NSHTTPURLResponse class]);
+  OCMStub([fakeResponse statusCode]).andReturn(200);
```

Comment:
nit: using `OCMExpect` + `OCMVerifyAll` or `OCMVerify` may improve coverage by validating that the expected methods have been actually called.

Reply:
Added verification to these tests in here - https://github.com/firebase/firebase-ios-sdk/pull/6138 which I will merge to this branch.